src/test/types/react-redux.js

```diff
@@ -88,6 +90,40 @@ const ConnectedExampleComponent = explicitConnect(
   }: ExplicitConnectOptions<OwnProps, StateProps, DispatchProps>)
 );
 
+{
+  // Test that WrapDispatchProps modifies the ThunkActions.
+  const wrapped: WrapDispatchProps<DispatchProps> = (ANY_VALUE: {|
+    +dispatchString: string => Action,
+    +dispatchThunk: string => number,
+  |});
+}
+
+{
+  // Test that the original unwrapped action creators do not work.
+  const wrapped: WrapDispatchProps<DispatchProps> = (ANY_VALUE: {|
+    +dispatchString: string => Action,
+    // $ExpectError
+    +dispatchThunk: string => ThunkAction<number>,
+  |});
+}
+
+{
+  // Test that WrapFunctionInDispatch works to strip off the return action.
+  const exampleAction = (string: string) => (ANY_VALUE: Action);
+  const exampleThunkAction = (string: string) => (
+    dispatch: Dispatch,
+    getState: GetState
+  ) => (ANY_VALUE: number);
+  const exampleThunkActionWrapped = (string: string) => 5;
+
+  (exampleAction: WrapFunctionInDispatch<ExampleActionCreator>);
+  (exampleThunkActionWrapped: WrapFunctionInDispatch<
+    ExampleThunkActionCreator
+  >);
+  // $ExpectError
+  (exampleThunkAction: WrapFunctionInDispatch<ExampleThunkActionCreator>);
+}
```

Comment:
can you add a line in the component to test the return value of an actual call to `dispatchThunk`? Something like 
```js
(this.props.dispatchThunk('foo'): number);
```